test/test_ops.py

```diff
@@ -227,17 +227,26 @@ def _ref_test_helper(self, ctx, device, dtype, op, skip_zero_numel=False, skip_z
                 )
             ):
                 continue
+
+            # Note: this deliberately uses 'op.torch_opinfo' in both cases
+            # because 'op.aliases' is empty.  When the aliases are executed
+            # within the context, they will behave as refs.  This is done to
+            # check for compatibility between all APIs.
+            aliases = op.torch_opinfo.aliases if op.test_aliases else ()
```

Comment:
doing `bar is foo` for refs might be tricky because of api incompat. see the example in the other comment for `special`. also feels less transparent for the users because it would rely on object ids.

i think it would be easier to have:
- the above alias test and
- to avoid duplicate work in the alias test, make sure users cannot register aliases when declaring refs in the `PythonRefInfo` constructor (https://github.com/pytorch/pytorch/pull/84502#discussion_r964717129)

  example: instead of having refs for `softmax`, `special.softmax`, and `nn.functional.softmax`, you'll just have one ref for `softmax` testing all the aliases.
  
the current plan is to have follow up PRs that would:
  - disallow declaring aliases as refs
  - maybe: add alias testing to tests where we'd lose coverage because of the above change (decomps?)
  - add missing refs for aliases (exposed by the current alias test, where `test_aliases=False` atm).
  
  wdyt?

Reply:
after thinking more about this, i think this might be better:
- i'll revert the alias testing part of this in the PR such that only operations (and aliases) are added here (i should have used ghstack for this, i know)
- i'll resubmit the alias testing part with the other testing-infra-related changes such that we can discuss and merge them all at once (instead of relying on follow up commits)

Reply:
Separating the aliasing part makes sense to me conceptually, I'll focus on reviewing the references